cve_bin_tool/output_engine/console.py

```diff
@@ -19,9 +19,37 @@
 from ..util import ProductInfo
 
 
+def parse_version_info(list_of_version):
+    STR_EMPTY = ""
+    RANGE_EMPTY = ["", ""]
+    affected_version = {"including": ["", ""], "excluding": ["", ""]}
+
+    if list_of_version:
+        # parse including versions
+        if list_of_version[0] is not STR_EMPTY:
+            affected_version["including"][0] = list_of_version[0]
+        if list_of_version[2] is not STR_EMPTY:
+            affected_version["including"][1] = list_of_version[2]
+
+        # parse excluding versions
+        if list_of_version[1] is not STR_EMPTY:
+            affected_version["excluding"][0] = list_of_version[1]
+        if list_of_version[3] is not STR_EMPTY:
+            affected_version["excluding"][1] = list_of_version[3]
+
+        if affected_version["including"] is not RANGE_EMPTY:
+            return f"[{affected_version['including'][0]} - {affected_version['including'][1]}]"
+        elif affected_version["excluding"] is not RANGE_EMPTY:
+            return f"({affected_version['excluding'][0]} - {affected_version['excluding'][1]})"
+    else:
+        return "-"
```

Comment:
1. `versions` is pretty self-explanatory, or maybe `list_of_versions`.
2. I'm pretty sure constant definitions in functions violate PEP8, you probably should make them regular variables (lowercase) or extract them outside of the function.
3. What's the point of `STR_EMPTY`? :) Just compare with an empty string :)
4. Pythonic way of checking an empty string is just `if str:`
5. All your default values for versions are empty strings and you change them only if they are not empty 🤔 I'm pretty sure all these `if`s are unnecessary!
6. `is not` in not the same as `!=`:
```py
s1 = "xyz"
s2 = "".join(["x", "y", "z"])
if s1 is not s2:
    print("Different")
else:
    print("Same")
```
This will print `Different` and it's probably not what you wanted 😉 

```suggestion
def parse_version_info(versions):
    empty_range = ["", ""]
    affected_versions = {
        "including": [versions[0], versions[2]],
        "excluding": [versions[1], versions[3]],
    }

    if versions:
        if affected_versions["including"] != empty_range:
            return f"[{affected_versions['including'][0]} - {affected_versions['including'][1]}]"
        if affected_versions["excluding"] != empty_range:
            return f"({affected_versions['excluding'][0]} - {affected_versions['excluding'][1]})"
    else:
        return "-"
```

Reply:
Thank you for the review! :D
That was a really clean hack.